Seems to be tied to the revision, the other problem. I invalidated to see if it gets it past
```
In file included from /data/civet1/build/blue_crab/build/unity_src/base_Unity.C:2:
/data/civet1/build/blue_crab/src/base/BlueCrabApp.C: In member function 'virtual std::string BlueCrabApp::header() const':
/data/civet1/build/blue_crab/src/base/BlueCrabApp.C:253:50: error: 'SODIUM_REVISION' was not declared in this scope; did you mean 'HELIUM_REVISION'?
  253 |   os << std::setw(25) << "  Sodium version: " << SODIUM_REVISION << "\n";
      |                                                  ^~~~~~~~~~~~~~~
      |                                                  HELIUM_REVISION
Compiling C++ (in opt mode) /data/civet1/build/blue_crab/build/unity_src/timesteppers_Unity.C...
Compiling C++ (in opt mode) /data/civet1/build/blue_crab/build/unity_src/postprocessors_Unity.C...
make: *** [/data/civet1/build/moose/framework/build.mk:150: /data/civet1/build/blue_crab/build/unity_src/base_Unity.x86_64-pc-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

ERROR: exiting with code 2
```

--- a/framework/app.mk
+++ b/framework/app.mk
@@ -225,7 +225,7 @@ app_EXEC    := $(APPLICATION_DIR)/$(APPLICATION_NAME)-$(METHOD)
 ifeq ($(GEN_REVISION),yes)
   CAMEL_CASE_NAME := $(shell echo $(APPLICATION_NAME) | perl -pe 's/(?:^|_)([a-z])/\u$$1/g')
   app_BASE_DIR    ?= base/
-  app_HEADER      ?= $(APPLICATION_DIR)/include/$(app_BASE_DIR)$(CAMEL_CASE_NAME)Revision.h
+  app_HEADER      := $(APPLICATION_DIR)/include/$(app_BASE_DIR)$(CAMEL_CASE_NAME)Revision.h
 endif
 
 # depend modules

--- a/modules/combined/Makefile
+++ b/modules/combined/Makefile
@@ -30,6 +30,7 @@ include $(MODULE_DIR)/modules.mk
 APPLICATION_DIR    := $(MODULE_DIR)/combined
 APPLICATION_NAME   := combined
 BUILD_EXEC         := yes
+GEN_REVISION       := yes
 # Sets INSTALLABLE_DIRS
 include            $(MOOSE_DIR)/modules/combined/combined_installable_dirs.mk
 include            $(FRAMEWORK_DIR)/app.mk

--- a/modules/fluid_properties/fluid_properties.mk
+++ b/modules/fluid_properties/fluid_properties.mk
@@ -19,6 +19,11 @@ POTASSIUM_FP_CONTENT      := $(shell ls $(POTASSIUM_FP_DIR) 2> /dev/null)
 SODIUM_FP_DIR             ?= ${MOOSE_DIR}/modules/fluid_properties/contrib/sodium
 SODIUM_FP_CONTENT         := $(shell ls $(SODIUM_FP_DIR) 2> /dev/null)
 
+# Cache revision information
+CAMEL_CASE_NAME_save := ${CAMEL_CASE_NAME}
+app_BASE_DIR_save := ${app_BASE_DIR}
+app_HEADER_save := ${app_HEADER}
+
 # AIR
 ifneq ($(AIR_FP_CONTENT),)
 ifneq ($(BUILDING_FP_APP), yes)
@@ -101,4 +106,12 @@ ifneq ($(BUILDING_FP_APP), yes)
 	include            $(FRAMEWORK_DIR)/app.mk
 	include            $(SODIUM_FP_DIR)/libSodiumProperties.mk
 endif
+endif
+
+# If building any fluid property submodule, we need to clear those before app.mk gets included
+# for any other module, as those modules do not generate a revision
+ifneq ($(AIR_FP_CONTENT)$(CARBON_DIOXIDE_FP_CONTENT)$(HELIUM_FP_CONTENT)$(NITROGEN_FP_CONTENT)$(POTASSIUM_FP_CONTENT)$(SODIUM_FP_CONTENT),)
+  CAMEL_CASE_NAME := ${CAMEL_CASE_NAME_save}
+	app_BASE_DIR := ${app_BASE_DIR_save}
+	app_HEADER := ${app_HEADER_save}
 endif

--- a/modules/fluid_properties/fluid_properties.mk
+++ b/modules/fluid_properties/fluid_properties.mk
@@ -23,6 +23,7 @@ SODIUM_FP_CONTENT         := $(shell ls $(SODIUM_FP_DIR) 2> /dev/null)
 CAMEL_CASE_NAME_save := ${CAMEL_CASE_NAME}
 app_BASE_DIR_save := ${app_BASE_DIR}
 app_HEADER_save := ${app_HEADER}
+GEN_REVISION_save := ${GEN_REVISION}
 
 # AIR
 ifneq ($(AIR_FP_CONTENT),)
@@ -114,4 +115,5 @@ ifneq ($(AIR_FP_CONTENT)$(CARBON_DIOXIDE_FP_CONTENT)$(HELIUM_FP_CONTENT)$(NITROG
   CAMEL_CASE_NAME := ${CAMEL_CASE_NAME_save}
 	app_BASE_DIR := ${app_BASE_DIR_save}
 	app_HEADER := ${app_HEADER_save}
+	GEN_REVISION := ${GEN_REVISION_save}
 endif

